### ISSUE TITLE
Service Check reports timeout before check_timeout exceeded

### ISSUE THREAD

**Issue by inmediasit** [Icinga/icinga2#7471]
## Describe the bug

We have some Windows 2008 R2 hosts where the command check_update.exe runs into a timeout. The monitoring check is pretty slow, it’s running for 5 to 10 minutes. After execution Icinga reports “Timeout exceeded”, but check_timeout is configured with a threshold of 15 minutes (900s).

The execution of the check command on the client host via Powershell dosn't report an error.

```
PS C:\Program Files\ICINGA2\sbin> Measure-Command { .\check_update.exe | Out-Host }
UPDATE CRITICAL 1 | 'update'=1;0;0;0;


Days              : 0
Hours             : 0
Minutes           : 9
Seconds           : 58
Milliseconds      : 758
Ticks             : 5987581476
TotalDays         : 0,00693007115277778
TotalHours        : 0,166321707666667
TotalMinutes      : 9,97930246
TotalSeconds      : 598,7581476
TotalMilliseconds : 598758,1476
```

## Expected behavior

The service check shouldn't run into a timeout and it should report the check result.

## Your Environment

Include as many relevant details about the environment you experienced the problem in

* Version used (`icinga2 --version`): r2.10.3-1
* Operating System and version: Debian GNU/Linux 9.9 (stretch)
* Enabled features (`icinga2 feature list`): api checker command ido-mysql mainlog notification

## Additional context

Maybe this issue is related to bug report #6992. That's why I also defined a 15 minutes check_timeout in the CheckCommand definition.

The object of the service check:

```
Object 'sbs.acme.net!Windows - Updates' of type 'Service':
  % declared in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/servicesets.conf', lines 125:1-125:33
  * __name = "sbs.acme.net!Windows - Updates"
  * action_url = ""
  * check_command = "update-windows"
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 171:5-171:36
  * check_interval = 1800
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 4:5-4:23
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 4:5-4:23
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 173:5-173:24
  * check_period = "24_7"
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 3:5-3:25
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 3:5-3:25
  * check_timeout = 900
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 6:5-6:22
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 6:5-6:22
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 175:5-175:23
  * command_endpoint = "sbs.acme.net"
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 13:5-13:32
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 13:5-13:32
  * display_name = "Windows - Updates"
  * enable_active_checks = true
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 8:5-8:31
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 8:5-8:31
  * enable_event_handler = true
  * enable_flapping = false
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 10:5-10:27
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 10:5-10:27
  * enable_notifications = true
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 7:5-7:31
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 7:5-7:31
  * enable_passive_checks = true
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 9:5-9:32
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 9:5-9:32
  * enable_perfdata = true
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 11:5-11:26
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 11:5-11:26
  * event_command = ""
  * flapping_threshold = 0
  * flapping_threshold_high = 30
  * flapping_threshold_low = 25
  * groups = [ "Windows Services" ]
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 27:5-27:35
  * host_name = "sbs.acme.net"
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/servicesets.conf', lines 125:1-125:33
  * icon_image = ""
  * icon_image_alt = ""
  * max_check_attempts = 3
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 2:5-2:28
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 2:5-2:28
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 172:5-172:28
  * name = "Windows - Updates"
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/servicesets.conf', lines 125:1-125:33
  * notes = "Eventuell weicht dieser Check vom Standardverhalten anderer Checks ab."
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 233:5-233:84
  * notes_url = "https://wiki.acme.net/index.php?title=Icinga_2#Sonderverhalten_von_Checks"
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 234:5-234:94
  * package = "director"
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/servicesets.conf', lines 125:1-125:33
  * retry_interval = 901
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 5:5-5:23
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 5:5-5:23
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 174:5-174:25
  * source_location
    * first_column = 1
    * first_line = 125
    * last_column = 33
    * last_line = 125
    * path = "/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/servicesets.conf"
  * templates = [ "Windows - Updates", "Windows - Updates", "WindowsService", "BaseService", "OverrideNotifications", "OverrideService", "BaseService", "host var overrides (Director)" ]
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/servicesets.conf', lines 125:1-125:33
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 167:1-167:36
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 24:1-24:33
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 1:0-1:29
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 237:1-237:40
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 230:1-230:34
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 1:0-1:29
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/001-director-basics.conf', lines 28:3-28:43
  * type = "Service"
  * vars
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/001-director-basics.conf', lines 41:7-41:51
    * notify = false
      % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 14:5-14:22
      % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 14:5-14:22
      % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 240:5-240:23
  * volatile = false
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 12:5-12:20
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/service_templates.conf', lines 12:5-12:20
  * zone = "acme.net"
    % = modified in '/var/lib/icinga2/api/packages/director/24a2725c-59fb-4148-a1b5-0c7aa73c32a1/zones.d/director-global/servicesets.conf', lines 125:1-125:33
```

**Comment by dnsmichi** [comment]
That's a known issue with check_timeout not available via command_endpoint cluster messages. I don't have the GH issue at hand, but I am aware of the problem.

**Comment by htriem** [comment]
Will be solved with #6992. Closing.
